src/dappsExplorer/utils.tsx

```diff
@@ -0,0 +1,24 @@
+import { DappV2WithCategoryNames } from 'src/dapps/types'
+
+/***
+ * Calculates a score based on the search term and dapp
+ * - +2 if the name matches the search term in a case sensitive way
+ * - +1 if the name matches the search term in a case insensitive way
+ * - +1 if the description matches the search term in a case insensitive way
+ * - +1 if the category matches the search term in a case insensitive way
+ * @param dapp: dapp with categoryNames
+ * @param searchTerm: the user input search string
+ * @returns {number} the score of the dapp 0 - 5
+ */
+export const calculateSearchScore = (dapp: DappV2WithCategoryNames, searchTerm: string) => {
```

Comment:
I think we are not considering the case where [the user inputs more than one word](https://www.figma.com/file/zcOyuTp5316yiguvL15as1?node-id=130:6385#351150969). I guess we have to increase the score for each matched word. Also consider if the exact searched term matched, in that case we should prioritize that match.